src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -1118,16 +1128,25 @@ def _get_authorization_code_worker(authority_url, resource, results):
         logger.warning('Authentication Error: Authorization code was not captured in query strings "%s"',
                        web_server.query_params)
         return
+
+    if 'state' in web_server.query_params:
+        state = web_server.query_params['state']
```

Comment:
Please do both `state` generating and checking inside this routine, so to verify the sanity as early as possible. Once matches, we can throw the state away right here and move on to acquire the token.